local.go

```diff
@@ -351,12 +353,19 @@ func (drive LocalDrive) listFolder(path string, prefix string, config *ListConfi
 
 	return res, nil
 }
-
-func (drive LocalDrive) checkName(path string) (string, error) {
+func (drive LocalDrive) checkName(path string, isFolder bool) (string, error) {
 	_, err := os.Stat(path)
 
 	for !os.IsNotExist(err) {
-		path = path + ".new"
+		ext := filepath.Ext(path)
+
+		if isFolder || ext == "" {
+			path = path + ".new"
+		} else {
+			index := strings.Index(path, ext)
```

Comment:
nope, index searches from the start of string, it possible that path already contains a similar pattern 

```index := len(path) - len(ext)```